lib/paginate.js

```diff
@@ -55,3 +55,31 @@ module.exports = {
     );
   },
 };
+
+/**
+ * Async generator to facilitate extremely large queries whose results
+ * cannot fit in memory all at once. The given query must accept an "offset"
+ * param that will be used to paginate the query.
+ *
+ * Should be used like this:
+ *
+ * for await (const row of paginateQueryAsync(sql, params)) {
+ *  // do something with row
+ * }
+ *
+ */
+module.exports.paginateQueryAsync = async function* (sql, params) {
+  const _params = Object.assign({}, params);
+  let offset = 0;
+  while (true) {
+    _params.offset = offset;
+    const result = await sqldb.queryAsync(sql, _params);
+    for (const row of result.rows) {
+      yield row;
+    }
```

Comment:
I don't have experience with `yield` in JS, but I was reading about it and this code may be simplified to (if I'm reading [this](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/yield*) correctly):
```suggestion
    yield* result.rows;
```

Reply:
Very nice, thanks!